test/CompanyService.Business.UnitTests/Commands/Department/CreateDepartmentCommandTests.cs

```diff
@@ -89,108 +66,188 @@ public void OneTimeSetUp()
+
+            _autoMock
+                .Setup<IHttpContextAccessor, IDictionary<object, object>>(x => x.HttpContext.Items)
+                .Returns(_items);
+
+            _autoMock
+                .Setup<IAccessValidator, bool>(x => x.IsAdmin(null))
                 .Returns(true);
+
+            _autoMock
+                .Setup<ICreateDepartmentRequestValidator, bool>(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                .Returns(true);
+
+            _autoMock
+                .Setup<ICompanyRepository, DbCompany>(x => x.Get(null))
+                .Returns(new DbCompany { Id = _companyId });
+
+            _autoMock
+                .Setup<IDbDepartmentMapper, DbDepartment>(x => x.Map(_request, _companyId))
+                .Returns(_dbDepartment);
+
+            _autoMock
+                .Setup<IDepartmentRepository, Guid>(x => x.CreateDepartment(_dbDepartment))
+                .Returns(_dbDepartment.Id);
+
+            _autoMock
+                .Setup<IDepartmentRepository, bool>(x => x.DoesNameExist(_request.Name))
+                .Returns(false);
         }
 
         [Test]
         public void ShouldThrowForbiddenExceptionWhenUserIsNotAdminAndNotEnoughRights()
         {
-            _accessValidatorMock
-                .Setup(x => x.IsAdmin(null))
+            _autoMock
+                .Setup<IAccessValidator, bool>(x => x.IsAdmin(null))
                 .Returns(false);
 
-            _accessValidatorMock
-                .Setup(x => x.HasRights(Rights.AddEditRemoveDepartments))
+            _autoMock
+                .Setup<IAccessValidator, bool>(x => x.HasRights(Rights.AddEditRemoveDepartments))
                 .Returns(false);
 
             Assert.Throws<ForbiddenException>(() => _command.Execute(_request));
-            _companyRepositoryMock.Verify(x => x.Get(It.IsAny<GetCompanyFilter>()), Times.Never);
-            _repositoryMock.Verify(x => x.CreateDepartment(It.IsAny<DbDepartment>()), Times.Never);
+
+            _autoMock.Verify<ICompanyRepository, DbCompany>(
+                x => x.Get(It.IsAny<GetCompanyFilter>()),
+                Times.Never);
+
+            _autoMock.Verify<IDepartmentRepository, Guid>(
+                x => x.CreateDepartment(It.IsAny<DbDepartment>()),
+                Times.Never);
+        }
+
+        [Test]
+        public void ShouldThrowValidationException()
+        {
+            _autoMock
+                .Setup<ICreateDepartmentRequestValidator, bool>(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                .Returns(false);
+
+            Assert.Throws<ValidationException>(() => _command.Execute(_request));
+
+            _autoMock.Verify<ICompanyRepository, DbCompany>(
+                x => x.Get(It.IsAny<GetCompanyFilter>()),
+                Times.Never);
+
+            _autoMock.Verify<IDepartmentRepository, Guid>(
+                x => x.CreateDepartment(It.IsAny<DbDepartment>()),
+                Times.Never);
```

Comment:
you can add verify for DoesNameExist and mapper

Reply:
done